packages/rest/src/router/metadata.ts

```diff
@@ -166,10 +168,18 @@ function resolveControllerSpec(
  * @param constructor Controller class
  */
 export function getControllerSpec(constructor: Function): ControllerSpec {
-  let spec = Reflector.getOwnMetadata(API_SPEC_KEY, constructor);
+  let spec = MetadataInspector.getClassMetadata<ControllerSpec>(
+    REST_API_SPEC_KEY,
+    constructor,
+    true,
+  );
   if (!spec) {
-    spec = resolveControllerSpec(constructor, spec);
-    Reflector.defineMetadata(API_SPEC_KEY, spec, constructor);
+    spec = resolveControllerSpec(constructor);
+    MetadataInspector.Reflector.defineMetadata(
```

Comment:
I wish we did not have to use low-level Reflector API here and leverage either class-level decorator factory or something like `MetadataInspector.setClassMetadata` (a new method I just made up now).

Not a big deal, the patch can be landed with the current implementation too.